web3/manager.py

```diff
@@ -146,10 +150,13 @@ def request_blocking(
         """
         Make a synchronous request using the provider
         """
-        response = self._make_request(method, params)
+        try:
+            response = self._make_request(method, params)
+        except eth_tester.exceptions.TransactionFailed as e:
```

Comment:
I think in the best place to do this would be inside of the `EthereumTesterProvider` somewhere in here:

https://github.com/ethereum/web3.py/blob/master/web3/providers/eth_tester/main.py#L85-L103

That keeps us from having to have `eth-tester` specific code mixed into the main library code.